test/HpackSpec.hs

```diff
@@ -94,3 +96,64 @@ spec = do
               old <- readFile file
               hpack `shouldReturn` outputUnchanged
               readFile file `shouldReturn` old
+
+    context "with an absolute path to local defaults" $ do
+      it "resolves the path correctly" $ do
+        withTempDirectory $ flip withCurrentDirectory $ do
```

Comment:
I'm dubious whether this tests any more than that given an absolute path `a`, `b </> a == a` for arbitrary `b`.  If we think we need to test something here, then I would prefer a unit test for `ensure`.